pkg/assets/secrets.go

```diff
@@ -0,0 +1,89 @@
+package assets
+
+import (
+	"bytes"
+	"encoding/base64"
+	"path/filepath"
+	"text/template"
+
+	"github.com/coreos/bootkube/pkg/assets/internal"
+)
+
+func NewCASecret(assets Assets) (Asset, error) {
+	var secret Asset
+
+	caCert, err := assets.Get(AssetPathCACert)
+	if err != nil {
+		return secret, err
+	}
+	caCert.Name = filepath.Base(caCert.Name)
+	caCert.Data = b64Byte(caCert.Data)
+
+	config := struct {
+		Name string
+		Data []Asset
+	}{
+		Name: "kube-ca",
+		Data: []Asset{caCert},
+	}
+
+	tmpl, err := template.New(config.Name).Parse(string(internal.SecretTemplate))
+	if err != nil {
+		return secret, err
+	}
+
+	var data bytes.Buffer
+	if err := tmpl.Execute(&data, config); err != nil {
+		return secret, err
+	}
+
+	secret.Name = AssetPathCASecret
+	secret.Data = data.Bytes()
+
+	return secret, nil
+}
+
+func NewAPIServerSecret(assets Assets) (Asset, error) {
+	var secret Asset
+
+	apiKey, err := assets.Get(AssetPathAPIServerKey)
+	if err != nil {
+		return secret, err
+	}
+	apiKey.Name = filepath.Base(apiKey.Name)
+	apiKey.Data = b64Byte(apiKey.Data)
+
+	apiCert, err := assets.Get(AssetPathAPIServerCert)
+	if err != nil {
+		return secret, err
+	}
+	apiCert.Name = filepath.Base(apiCert.Name)
+	apiCert.Data = b64Byte(apiCert.Data)
+
+	config := struct {
```

Comment:
I think I'm gonna drop this code and just create a secret struct & marshal directly. Too much weirdness here for not much benefit.